parsers/ENTSOE.py

```diff
@@ -1062,66 +1092,15 @@ def fetch_production(
     if not session:
         session = Session()
     domain = ENTSOE_DOMAIN_MAPPINGS[zone_key]
-    # Grab production
-    parsed = parse_production(
-        query_production(domain, session, target_datetime=target_datetime)
-    )
-
-    if not parsed:
+    raw_production = query_production(domain, session, target_datetime=target_datetime)
+    if raw_production is None:
         raise ParserException(
             parser="ENTSOE.py",
             message=f"No production data found for {zone_key}",
             zone_key=zone_key,
         )
-
-    productions, production_dates = parsed
-
-    data = []
-    for i in range(len(production_dates)):
-        production_values = {k: v for k, v in productions[i].items()}
-        production_date = production_dates[i]
-
-        production_types = {"production": {}, "storage": {}}
-        for key in ["production", "storage"]:
-            parameter_groups = ENTSOE_PARAMETER_GROUPS[key]
-            multiplier = -1 if key == "storage" else 1
-
-            for fuel, groups in parameter_groups.items():
-                has_value = any(
-                    [production_values.get(grp) is not None for grp in groups]
-                )
-                if has_value:
-                    value = sum([production_values.get(grp, 0) for grp in groups])
-                    value *= multiplier
-                else:
-                    value = None
-
-                production_types[key][fuel] = value
-
-        data.append(
-            {
-                "zoneKey": zone_key,
-                "datetime": production_date,
-                "production": production_types["production"],
-                "storage": {
-                    "hydro": production_types["storage"]["hydro storage"],
-                },
-                "source": "entsoe.eu",
-            }
-        )
-
-        for d in data:
-            for k, v in d["production"].items():
-                if v is None:
-                    continue
-                if v < 0 and v > -50:
-                    # Set small negative values to 0
-                    logger.warning(
-                        "Setting small value of %s (%s) to 0." % (k, v),
-                        extra={"key": zone_key},
-                    )
-                    d["production"][k] = 0
-
+    parsed = parse_production(raw_production, logger, zone_key)
+    data = parsed.to_list()
```

Comment:
Looks like somewhere along the line we convert the datetime objects to timestamps, which don't work with the test_parser at least.

Reply:
Has been fixed with #5542